src/dune_engine/load_rules.ml

```diff
@@ -139,18 +139,18 @@ let describe_rule (rule : Rule.t) =
   Pp.text
   @@
   match rule.info with
-  | From_dune_file loc ->
-    let start = Loc.start loc in
+  | From_dune_file ->
+    let start = Loc.start rule.loc in
     start.pos_fname ^ ":" ^ string_of_int start.pos_lnum
   | Internal -> "<internal location>"
-  | Source_file_copy _ -> "file present in source tree"
+  | Source_file_copy -> "file present in source tree"
 ;;
 
 let report_rule_src_dir_conflict dir fn (rule : Rule.t) =
   let loc =
     match rule.info with
-    | From_dune_file loc -> loc
-    | Internal | Source_file_copy _ ->
+    | From_dune_file -> rule.loc
+    | Internal | Source_file_copy ->
```

Comment:
If we have `rule.loc`, why do we look at `rule.info` to come up with some other `loc` here?